lib/msf/core/payload_generator.rb

```diff
@@ -429,12 +429,16 @@ def get_encoders
         encoder.split(',').each do |chosen_encoder|
           e = framework.encoders.create(chosen_encoder)
           if e.nil?
-            cli_print "Skipping invalid encoder #{chosen_encoder}"
+            cli_print "[-] Skipping invalid encoder #{chosen_encoder}"
             next
           end
           e.datastore.import_options_from_hash(datastore)
           encoders << e if e
         end
+        if encoders.empty?
+          cli_print "[!] Couldn't find encoder to use"
+          exit(1)
```

Comment:
I think you should `return` from the method, not `exit` the process? This is in the library and not the command-line tool.

Reply:
As long as it doesn't try to keep on running (like it did before), I don't mind altering it =)